.github/workflows/weekly_update.yml

```
@@ -5,6 +5,10 @@ on:
     - cron: "0 0 * * 1"  # every Monday at midnight
   workflow_dispatch:
 
+permissions:
+  contents: write
+  pull-requests: write
```

As mentioned in the PR description, another alternative would be
```suggestion
permissions: write-all
```